File: .sdk-repos/orchestration-cluster-api-rust/examples/cluster_variable.rs
//! Compilable usage examples for cluster variable operations.
//!
//! Region-tagged snippets are surfaced in the Camunda REST API documentation via
//! `examples/operation-map.json`, and are type-checked by `cargo build --examples`.
#![allow(dead_code, unused_variables, unused_imports)]

use camunda_orchestration_sdk::apis::cluster_variable_api::CreateGlobalClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::CreateTenantClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::DeleteGlobalClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::DeleteTenantClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::GetGlobalClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::GetTenantClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::SearchClusterVariablesParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::UpdateGlobalClusterVariableParams;
use camunda_orchestration_sdk::apis::cluster_variable_api::UpdateTenantClusterVariableParams;
use camunda_orchestration_sdk::models::ClusterVariableName;
use camunda_orchestration_sdk::models::ClusterVariableSearchQueryRequest;
use camunda_orchestration_sdk::models::CreateClusterVariableRequest;
use camunda_orchestration_sdk::models::UpdateClusterVariableRequest;
use camunda_orchestration_sdk::CamundaClient;

// region CreateGlobalClusterVariable
async fn create_global_cluster_variable() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .create_global_cluster_variable(CreateGlobalClusterVariableParams {
            create_cluster_variable_request: CreateClusterVariableRequest {
                name: ClusterVariableName::assume_exists("my-variable"),
                value: serde_json::json!({"key": "value"}),
                ..Default::default()
            },
        })
        .await?;
    println!("{}", result.name);

    Ok(())
}
// endregion CreateGlobalClusterVariable

// region CreateTenantClusterVariable
async fn create_tenant_cluster_variable(
    tenant_id: String,
) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .create_tenant_cluster_variable(CreateTenantClusterVariableParams {
            tenant_id,
            create_cluster_variable_request: CreateClusterVariableRequest {
                name: ClusterVariableName::assume_exists("my-variable"),
                value: serde_json::json!({"key": "value"}),
                ..Default::default()
            },
        })
        .await?;
    println!("{}", result.name);

    Ok(())
}
// endregion CreateTenantClusterVariable

// region DeleteGlobalClusterVariable
async fn delete_global_cluster_variable(name: String) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_global_cluster_variable(DeleteGlobalClusterVariableParams { name })
        .await?;
    println!("Delete a global-scoped cluster variable: done");

    Ok(())
}
// endregion DeleteGlobalClusterVariable

// region DeleteTenantClusterVariable
async fn delete_tenant_cluster_variable(
    tenant_id: String,
    name: String,
) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .delete_tenant_cluster_variable(DeleteTenantClusterVariableParams { tenant_id, name })
        .await?;
    println!("Delete a tenant-scoped cluster variable: done");

    Ok(())
}
// endregion DeleteTenantClusterVariable

// region GetGlobalClusterVariable
async fn get_global_cluster_variable(name: String) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .get_global_cluster_variable(GetGlobalClusterVariableParams { name })
        .await?;
    println!("{}", result.name);

    Ok(())
}
// endregion GetGlobalClusterVariable

// region GetTenantClusterVariable
async fn get_tenant_cluster_variable(
    tenant_id: String,
    name: String,
) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .get_tenant_cluster_variable(GetTenantClusterVariableParams { tenant_id, name })
        .await?;
    println!("{}", result.name);

    Ok(())
}
// endregion GetTenantClusterVariable

// region SearchClusterVariables
async fn search_cluster_variables() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .search_cluster_variables(SearchClusterVariablesParams {
            truncate_values: None,
            cluster_variable_search_query_request: Some(
                ClusterVariableSearchQueryRequest::default(),
            ),
        })
        .await?;
    for item in result.items {
        println!("{}", item.name);
    }

    Ok(())
}
// endregion SearchClusterVariables

// region UpdateGlobalClusterVariable
async fn update_global_cluster_variable(name: String) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .update_global_cluster_variable(UpdateGlobalClusterVariableParams {
            name,
            update_cluster_variable_request: UpdateClusterVariableRequest {
                value: serde_json::json!({"key": "value"}),
                ..Default::default()
            },
        })
        .await?;
    println!("{}", result.name);

    Ok(())
}
// endregion UpdateGlobalClusterVariable

// region UpdateTenantClusterVariable
async fn update_tenant_cluster_variable(
    tenant_id: String,
    name: String,
) -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let result = client
        .update_tenant_cluster_variable(UpdateTenantClusterVariableParams {
            tenant_id,
            name,
            update_cluster_variable_request: UpdateClusterVariableRequest {
                value: serde_json::json!({"key": "value"}),
                ..Default::default()
            },
        })
        .await?;
    println!("{}", result.name);

    Ok(())
}
// endregion UpdateTenantClusterVariable

fn main() {
    // Examples above are compiled, not executed.
}

File: .sdk-repos/orchestration-cluster-api-php/examples/cluster_variable.php
<?php

/**
 * Compilable usage examples for global and tenant cluster-variable operations.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;

// region CreateGlobalClusterVariable
/**
 * Create a global-scoped cluster variable.
 */
function create_global_cluster_variable(CamundaClient $client, \Camunda\Orchestration\Api\Model\CreateClusterVariableRequest $createClusterVariableRequest): void
{
    $client->createGlobalClusterVariable($createClusterVariableRequest);
}
// endregion CreateGlobalClusterVariable

// region GetGlobalClusterVariable
/**
 * Get a global-scoped cluster variable.
 */
function get_global_cluster_variable(CamundaClient $client, string $name): void
{
    $client->getGlobalClusterVariable($name);
}
// endregion GetGlobalClusterVariable

// region UpdateGlobalClusterVariable
/**
 * Update a global-scoped cluster variable.
 */
function update_global_cluster_variable(CamundaClient $client, string $name, \Camunda\Orchestration\Api\Model\UpdateClusterVariableRequest $updateClusterVariableRequest): void
{
    $client->updateGlobalClusterVariable($name, $updateClusterVariableRequest);
}
// endregion UpdateGlobalClusterVariable

// region DeleteGlobalClusterVariable
/**
 * Delete a global-scoped cluster variable.
 */
function delete_global_cluster_variable(CamundaClient $client, string $name): void
{
    $client->deleteGlobalClusterVariable($name);
}
// endregion DeleteGlobalClusterVariable

// region SearchClusterVariables
/**
 * searchClusterVariables.
 */
function search_cluster_variables(CamundaClient $client, ?bool $truncateValues = null, ?\Camunda\Orchestration\Api\Model\ClusterVariableSearchQueryRequest $clusterVariableSearchQueryRequest = null): void
{
    $client->searchClusterVariables($truncateValues, $clusterVariableSearchQueryRequest);
}
// endregion SearchClusterVariables

// region CreateTenantClusterVariable
/**
 * Create a tenant-scoped cluster variable.
 */
function create_tenant_cluster_variable(CamundaClient $client, string $tenantId, \Camunda\Orchestration\Api\Model\CreateClusterVariableRequest $createClusterVariableRequest): void
{
    $client->createTenantClusterVariable($tenantId, $createClusterVariableRequest);
}
// endregion CreateTenantClusterVariable

// region GetTenantClusterVariable
/**
 * Get a tenant-scoped cluster variable.
 */
function get_tenant_cluster_variable(CamundaClient $client, string $tenantId, string $name): void
{
    $client->getTenantClusterVariable($tenantId, $name);
}
// endregion GetTenantClusterVariable

// region UpdateTenantClusterVariable
/**
 * Update a tenant-scoped cluster variable.
 */
function update_tenant_cluster_variable(CamundaClient $client, string $tenantId, string $name, \Camunda\Orchestration\Api\Model\UpdateClusterVariableRequest $updateClusterVariableRequest): void
{
    $client->updateTenantClusterVariable($tenantId, $name, $updateClusterVariableRequest);
}
// endregion UpdateTenantClusterVariable

// region DeleteTenantClusterVariable
/**
 * Delete a tenant-scoped cluster variable.
 */
function delete_tenant_cluster_variable(CamundaClient $client, string $tenantId, string $name): void
{
    $client->deleteTenantClusterVariable($tenantId, $name);
}
// endregion DeleteTenantClusterVariable
